packages/browser-wallet/src/popup/pages/Account/TransactionLog/TransactionList.tsx

```diff
@@ -291,16 +312,15 @@ export default function TransactionList({ onTransactionClick }: TransactionListP
             );
         }
     } else {
-        // If the first transaction in the list changes, then we need to reload the entire
-        // list, otherwise the infinite list component bugs out displaying the transactions
-        // correctly.
-        const listKey = transactions[0].transactionHash ?? transactions[0].id;
+        // Quick and dirty fingerprint of transactions, to ensure transaction list resets properly.
+        const listKey = `${allTransactions.length}${allTransactions.reduce((acc, cur) => (acc + cur.id) % 73, 0)}`;
```

Comment:
I would think that the previous key prepended with the account address should work instead of this.

Reply:
What do you mean with the previous key? the key from the previous render?

I was getting lists that didn't properly reset their sizes due to the key not resetting properly. As far as I remember, this was only really an issue when changing between accounts which both had the same latest transaction in their list, so a combination of the latest transaction and the currently selected account is not sufficient.

Reply:
I think he meant accountAddress + (transactions[0].transactionHash ?? transactions[0].id)

Reply:
But why would that not work, when switching between accounts with the same latest transcaction, the address would change, causing a rerender, right?

Reply:
No, because the transactions are fetched as a result of the account changing, meaning the following would happen:

- account changes, re-render list with key `{address}{pendingTxHash}`.
- get new transactions, list key remains the same, due to pending transactions not resetting (which is the case with remote transactions)

Reply:
But I guess i could ignore the pending transactions, and make the key based on the first remote transaction.

Reply:
I don't understand this still. Doesn't the pending transactions also change? (Why are they not resetting?)

Reply:
Is it just the case where you switch between the sender and recipient? :thinking:

Reply:
You changed it :thinking: